op-node/rollup/interop/managed/system.go

```diff
@@ -378,60 +378,131 @@ func (m *ManagedMode) Reset(ctx context.Context, lUnsafe, xUnsafe, lSafe, xSafe,
 	return nil
 }
 
-// findLatestValidLocalUnsafe scans the L2 unsafe chain for the latest block with a valid L1 origin,
-// starting from its latest unsafe and going back until the target provided by the supervisor.
-// If the supervisor's target is ahead, the latest unsafe is returned.
+// findLatestValidLocalUnsafe searches and returns the latest valid block of the L2 chain
+// starting from `l2UnsafeTarget` and checking until the latest unsafe block.
 func (m *ManagedMode) findLatestValidLocalUnsafe(ctx context.Context, l2UnsafeTarget eth.BlockID) (eth.L2BlockRef, error) {
 	latestUnsafe, err := m.l2.L2BlockRefByLabel(ctx, eth.Unsafe)
 	if err != nil {
 		return eth.L2BlockRef{}, err
 	}
+
 	logger := m.log.New("target", l2UnsafeTarget, "latestUnsafe", latestUnsafe)
-	if latestUnsafe.Number < l2UnsafeTarget.Number {
```

Comment:
I am modifying the behavior here. In this case we will still check if `l2UnsafeTarget` is valid, rather than just return it.